lib/rib/extra/autoindent.rb

```diff
@@ -35,7 +35,8 @@ module Rib; module Autoindent
     /^if\b/            => /^(end)\b|^else$|^elsif\b/,
     /^unless\b/        => /^(end)\b|^else$|^elsif\b/,
     /^case\b/          => /^(end)\b|^else$|when\b/  ,
-    /^def\b/           => /^(end)\b/                ,
+    /^def\b/           =>
+      /^(end)\b|^else$|^rescue *((\w+)? *(=> *\w+)?)?$|^ensure$/,
```

Comment:
I'll move this section right under `begin` so that it's clear they're the same after I merge this (so we save some round trip :)